Automattic/jetpack
# Coding Standards: Add CS Package

When writing up some documentation for a new filter in 8.5, I found a hook docblock that had the incorrect version number (`8.5` vs `8.5.0`) and wanted to prevent that from happening in the future.

This is a first pass at a Coding Standards package for Jetpack.

It still needs unit tests and, ideally, PHPCS-compliant docs.

#### Changes proposed in this Pull Request:
* Modifies the Jetpack .phpcs.xml.dist for jetpack repo-specific needs.
* Excludes the package from some WPCS rules to match PHPCS style.
* Creates a "Jetpack" ruleset that can be included by other projects via Composer.
* The Jetpack ruleset is the existing WP coding standards, with the following additions:
** Verifies that hook execution function calls (`apply_filters`, `do_action`, etc) are preceded by a docblock comment.
** Verifies that the docblock comment has a `@since` tag and a `@module` tag.

I believe there are plenty of places in Jetpack where we don't assign a module to a filter, so that is something we can remove.

The `@since` enforcement will force a X.Y.Z version while still allow an annotation following it (e.g. both `@since X.Y.Z` and `@since X.Y.Z Changes default` are allowed).

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Adds a new package. Not discussed on a p2.

#### Testing instructions:
* `composer install`
* Add `apply_filters` or one of the other hook execution functions. Ensure the tests flag lack of docblock.
* Add a docblock with various levels of completeness. Confirm it flags it.
* Add a `/** This filter is documented in XYZ */` style line. Confirm it passes.

#### Proposed changelog entry for your changes:
* Package: Adds a Jetpack Coding Standards package.
